packages/synapse-react-client/src/utils/hooks/useCookiePreferences.ts

```diff
@@ -0,0 +1,45 @@
+import { useCallback } from 'react'
+import { atom, useAtomValue, useSetAtom } from 'jotai'
+
+export const getCurrentCookiePreferences = () => {
+  const prefs = localStorage.getItem(COOKIES_AGREEMENT_LOCALSTORAGE_KEY)
+  const allowNone: CookiePreference = {
+    functionalAllowed: false,
+    analyticsAllowed: false,
+  }
+  return prefs != null ? (JSON.parse(prefs) as CookiePreference) : allowNone
+}
+export const COOKIES_AGREEMENT_LOCALSTORAGE_KEY =
+  'org.sagebionetworks.security.cookies.portal.preference'
+
+export type CookiePreference = {
+  functionalAllowed: boolean
+  analyticsAllowed: boolean
+}
+
+const cookiePreferencesAtom = atom<CookiePreference>(
+  getCurrentCookiePreferences(),
+)
```

Comment:
It makes sense to me to use Jotai here since it gives you global state. You could alternatively use `useLocalStorageValue` from `@react-hookz/web` ([docs](https://react-hookz.github.io/web/?path=/docs/side-effect-uselocalstoragevalue--example)) which might consolidate/abstract a lot of this logic